### PR TITLE
Add support for virtual machine disks

### PR DESCRIPTION
**Pull request by svanharmelen**

### CODE REVIEW

File: management/virtualmachinedisk/client.go [hunk 1]
```diff
@@ -17,15 +27,207 @@ func NewClient(client management.Client) DiskClient {
...
+	}
+
+	return c.client.SendAzurePostRequest(requestURL, req)
+}
+
+// AddDisk adds an operating system disk or data disk to the user image repository
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157178.aspx
+func (c DiskClient) AddDisk(params CreateDiskParameters) (management.OperationID, error) {
+	requestURL := fmt.Sprintf(addDiskURL)
```
Comment (paulmey) [inline management/virtualmachinedisk/client.go]: Why not just `requestURL := addDiskURL`? Or why have requestURL here at all?
I'd just use addDiskURL in `SendAzurePostRequest` below...

Comment (svanharmelen) [inline management/virtualmachinedisk/client.go]: Yeah, sure... I think just to be inline with the rest, but your right. Will update if you want me to :wink:

File: management/virtualmachinedisk/client.go [hunk 2]
```diff
@@ -17,15 +27,207 @@ func NewClient(client management.Client) DiskClient {
...
+}
+
+// ListDisks retrieves a list of the disks in the image repository that is associated
+// with the specified subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/jj157176.aspx
+func (c DiskClient) ListDisks() (ListDiskResponse, error) {
+	var response ListDiskResponse
+
+	requestURL := fmt.Sprintf(listDisksURL)
```
Comment (paulmey) [inline management/virtualmachinedisk/client.go]: Same, I'd skip on `requestURL` here and just use `listDisksURL` in `SendAzureGetRequest` below...

**Comment by paulmey** [comment]
LGTM, we'll start working these into the integration tests when merged.

File: management/virtualmachinedisk/client.go (hunk 1 above)
Comment (paulmey) [inline management/virtualmachinedisk/client.go]: Go ahead and update. I'll give Ahmet a bit more time to cast a veto before merging anyway...

File: management/virtualmachinedisk/client.go (hunk 2 above)
Comment (svanharmelen) [inline management/virtualmachinedisk/client.go]: Ok... Give me a sec to update those two then...

**Comment by svanharmelen** [comment]
Updated according to your feedback and squashed the commits...

**Comment by paulmey** [comment]
Thanks, I'll merge later this afternoon, when I get back to a machine...

Sent from my Windows Phone
________________________________

**Comment by svanharmelen** [comment]
Cool... Thx...

**Comment by paulmey** [comment]
Thanks @svanharmelen

**Comment by ahmetalpbalkan** [comment]
This is great work, thanks indeed @svanharmelen

**Pull request merged**